pkg/ciliumenvoyconfig/cec_resource_parser.go

```diff
@@ -553,6 +559,40 @@ func injectCiliumL7Filter(hcmConfig *envoy_config_http.HttpConnectionManager) bo
 	return false
 }
 
+func qualifyHttpFilters(cecNamespace string, cecName string, hcmConfig *envoy_config_http.HttpConnectionManager) bool {
+	updated := false
+
+	for _, httpFilter := range hcmConfig.HttpFilters {
+		switch h := httpFilter.ConfigType.(type) {
+		case *envoy_config_http.HttpFilter_TypedConfig:
+			any, err := h.TypedConfig.UnmarshalNew()
+			if err != nil {
+				continue
+			}
+
+			switch httpFilterConfig := any.(type) {
+			case *envoy_config_healthcheck.HealthCheck:
+				clusters := map[string]*envoy_config_types.Percent{}
+				for c, p := range httpFilterConfig.ClusterMinHealthyPercentages {
+					clusterName := c
+					updatedClusterName, nameUpdated := api.ResourceQualifiedName(cecNamespace, cecName, c)
+					if nameUpdated {
+						updated = true
+						clusterName = updatedClusterName
+					}
+
+					clusters[clusterName] = p
+				}
+				httpFilterConfig.ClusterMinHealthyPercentages = clusters
+
+				h.TypedConfig = toAny(httpFilterConfig)
```

Comment:
These two lines should only be executed if `updated == true`

Reply:
good point - thanks 

-> changed